packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -357,14 +358,16 @@ export class ApolloServerBase {
     // TODO: This is a bit nasty because the subscription server needs this.schema synchronously, for reasons of backwards compatibility.
     const _schema = this.initSchema();
 
-    if (_schema instanceof GraphQLSchema) {
+    if (isSchema(_schema)) {
       const derivedData = this.generateSchemaDerivedData(_schema);
       this.schema = derivedData.schema;
       this.schemaDerivedData = Promise.resolve(derivedData);
-    } else {
+    } else if (_schema.then) {
```

Comment:
Guarding if this isn't a function seems like it could avoid a 'not a function' error upon invocation of... not a function!:

```suggestion
    } else if (typeof _schema.then === 'function') {
```